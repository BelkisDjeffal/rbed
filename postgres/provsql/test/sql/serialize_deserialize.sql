\set ECHO none
SET search_path TO provsql_test, provsql;

CREATE TABLE test(id varchar PRIMARY KEY, v integer);

INSERT INTO test(id,v) VALUES
	('hello',4),
	('hi',8);

SELECT add_provenance('test');

do $$begin
PERFORM set_prob(provenance(), 0.2) from test where id = 'hello';
PERFORM set_prob(provenance(), 0.3) from test where id = 'hi';
end $$;


SELECT dump_data();
do $$ begin
PERFORM set_prob(provenance(), 0.4) from test where id = 'hello';
end $$;


SELECT read_data_dump();

CREATE TABLE result AS SELECT get_prob(provenance()) FROM test;
SELECT remove_provenance('result');
SELECT * FROM result;
DROP TABLE result;
DROP TABLE test;
