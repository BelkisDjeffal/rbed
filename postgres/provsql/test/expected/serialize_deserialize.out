\set ECHO none
 add_provenance 
----------------
 
(1 row)

INFO:  serializing completed without error
 dump_data 
-----------
 
(1 row)

INFO:  deserialization completed without error
 read_data_dump 
----------------
 
(1 row)

 remove_provenance 
-------------------
 
(1 row)

 get_prob 
----------
      0.2
      0.3
(2 rows)

